main/models.py

```diff
@@ -1,8 +1,21 @@
 """Models module for the main app."""
 
+from typing import ClassVar
+
 from django.contrib.auth.models import AbstractUser
 from django.db import models  # noqa: F401
 
 
 class User(AbstractUser):
     """Custom user model for this project."""
+
+    class Meta:
+        """Meta class for the User model."""
+
+        permissions: ClassVar = [
+            ("can_boot_processes", "Can boot processes"),
+            ("can_restart_processes", "Can restart processes"),
+            ("can_flush_processes", "Can flush processes"),
+            ("can_kill_processes", "Can kill processes"),
+            ("can_view_process_logs", "Can view process logs"),
+        ]
```

Comment:
Do we need all these separate permissions? I can see you *might* want some users to only be able to view the logs, but does it make sense to have users who can kill and boot processes, but not restart them?

Reply:
How about one for `view_logs` and one for `modify_processes`?

Reply:
That makes sense to me: kind of a read vs write distinction.